src/lib/span.ts

```diff
@@ -0,0 +1,38 @@
+export class Span {
+    spanId: string;
+    name: string;
+    startTime: number;
+    endTime: number | null;
+    parentSpanId: string | null;
+    children: Span[];
+
+    constructor(name: string, parentSpanId: string | null = null) {
+        this.spanId = `span-${Date.now()}-${Math.random().toString(36).substr(2, 5)}`;
+        this.name = name;
+        this.startTime = Date.now();
+        this.endTime = null;
+        this.parentSpanId = parentSpanId;
+        this.children = [];
+    }
+
```

Comment:
The only things that need to be set in the constructor are things that need the be passed through the constructor. Any other property can be defaulted when defined above with `public startTime: number = Date.now()`

The `spanId` should be an `IdSortable` from `@matrixai/id` library.